example_unrolling_service/example_without_bazel.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""This script demonstrates how the Python example service without needing
+to use the bazel build system.
+Prerequisite:
+    # In the repo's INSTALL.md, follow the 'Building from source using CMake' instructions with `-DCOMPILER_GYM_BUILD_EXAMPLES=ON` added to the `cmake` command
+    $ cd <path to source directory>/examples
+Usage:
+    $ python example_unrolling_service/examples_without_bazel.py
+It is equivalent in behavior to the example.py script in this directory.
+"""
+import os
+import subprocess
+from pathlib import Path
+from typing import Iterable
+
+import compiler_gym
+from compiler_gym.datasets import Benchmark, Dataset
+from compiler_gym.envs.llvm.llvm_benchmark import get_system_includes
+from compiler_gym.spaces import Reward
+from compiler_gym.third_party import llvm
+from compiler_gym.util.registration import register
+from compiler_gym.util.commands import run_command
+
+UNROLLING_PY_SERVICE_BINARY: Path = Path(
+    "example_unrolling_service/service_py/example_service.py"
+)
+assert UNROLLING_PY_SERVICE_BINARY.is_file(), "Service script not found"
+
+BENCHMARKS_PATH: Path = Path("benchmarks")
+
+NEURO_VECTORIZER_HEADER: Path = Path(
+    "third_party/neuro-vectorizer/header.h"
+)
+
+
+class RuntimeReward(Reward):
+    """An example reward that uses changes in the "runtime" observation value
+    to compute incremental reward.
+    """
+
+    def __init__(self):
+        super().__init__(
+            id="runtime",  # name="runtime", #FIXME
+            observation_spaces=["runtime"],
+            default_value=0,
+            default_negates_returns=True,
+            deterministic=False,
+            platform_dependent=True,
+        )
+        self.baseline_runtime = 0
+
+    def reset(self, benchmark: str, observation_view):
+        del benchmark  # unused
+        self.baseline_runtime = observation_view["runtime"]
+
+    def update(self, action, observations, observation_view):
+        del action  # unused
+        del observation_view  # unused
+        return float(self.baseline_runtime - observations[0]) / self.baseline_runtime
+
+
+class SizeReward(Reward):
+    """An example reward that uses changes in the "size" observation value
+    to compute incremental reward.
+    """
+
+    def __init__(self):
+        super().__init__(
+            id="size",  # name="size", #FIXME
```

Comment:
Thanks for bringing this one to my attention! I've added a workaround so that the old Reward.id won't break when we next push a release: https://github.com/facebookresearch/CompilerGym/pull/612